src/openshift/openshiftItem.ts

```diff
@@ -32,16 +32,42 @@ export abstract class OpenShiftItem {
         return new Promise((res) => setTimeout(res, timeout));
     }
 
+    static validateUniqueName(data: Array<OpenShiftObject>, value: string) {
+        const openshiftObject =  data.find((openshiftObject) =>  openshiftObject.getName() === value);
+        return openshiftObject && `This name is already used, please enter different name.`;
+    }
+
+    static async getName(message: string, data: Array<OpenShiftObject>): Promise<string> {
+        return await window.showInputBox({
+            prompt: `Provide ${message}`,
+            validateInput: (value: string) => {
+                let validationMessage = OpenShiftItem.emptyName(`Empty ${message}`, value.trim());
+                if (!validationMessage) validationMessage = OpenShiftItem.validateMatches(`Not a valid ${message}. Please use lower case alphanumeric characters or "-", and must start and end with an alphanumeric character`, value);
+                if (!validationMessage) validationMessage = OpenShiftItem.lengthName(`${message} is too long`, value);
```

Comment:
It should work for projects, applications ans services, but not components. The reason is a component name entered by user then combined with application name when actual deployment config created.
so getName function should have additional parameter than would offset 63 by length of  '${application-name}-}, default value would be 0.